### PR TITLE
Improved "cannot find API" error message

### PR DESCRIPTION
**Pull request by Erotemic**
I encountered an error when trying to use the `labelme` software. It was totally my fault, but I would have been able to figure that out sooner if the error message was more clear. 

It turned out that I had the `export QT_API=pyqt` environment variable in my bashrc (which I must have put in years ago and forgot about it). However, I have a modern python3 + pyqt5 environment setup. The init file tried to resolve to PyQt4. The error message I got was: "No Qt bindings could be found", which was confusing because I had pyqt5 installed. It also spit out an error about pyside because it tried to fall back on that. 

Unfortunately, because the PyQt4 code is after the PyQt5 code, it never tried to fallback on pyqt5 (another potential place for improvement).  

I've updated the init to warn each time it fails to fallback on a different API, and I also updated the final error message to indicate if the user has explicitly set the QT_API environment variable (and suggest that they may want to change it). 

Anyway, I hope this helps any other unfortunate soul that finds themself in my situation.

### CODE REVIEW

**Comment by goanpeca** [comment]
@Erotemic thanks for working on this!

**Comment by ccordoba12** [comment]
@Erotemic, thanks for your contribution! This now needs a merge with master to fix a merge conflict.

File: qtpy/__init__.py [hunk 1]
```diff
@@ -100,12 +100,21 @@ class PythonQtWarning(Warning):
 # Names of the expected PySide2 api
 PYSIDE2_API = ['pyside2']
 
+VALID_APIS = PYQT5_API + PYQT4_API + PYSIDE_API + PYSIDE2_API
+
+# Record if an explicit version of Qt was requested, so we can report this if
+# an error occurs.
+_EXPLICIT_API = None
+if QT_API in os.environ:
+    _EXPLICIT_API = os.environ[QT_API]
```
Comment (goanpeca) [inline qtpy/__init__.py]: Are there any other variables names that could have been used? or is this exclusive?

Comment (Erotemic) [reply qtpy/__init__.py]: I'm don't quite understand the question, could you rephrase?

Comment (goanpeca) [reply qtpy/__init__.py]: @Erotemic what I wat to say if QT Docs only talk about `QT_API` as an environment variable, or are there other variants?

Comment (ccordoba12) [reply qtpy/__init__.py]: I think `QT_API` was defined by the Python community (probably by Matplotlib or IPython).

Comment (goanpeca) [reply qtpy/__init__.py]: Hmm ok :-), then it should be fine

**Comment by goanpeca** [comment]
Ready to merge @ccordoba12 ?

File: qtpy/__init__.py [hunk 2]
```diff
@@ -148,7 +162,9 @@ class PythonQtWarning(Warning):
 
             del macos_version
     except ImportError:
-        API = os.environ['QT_API'] = 'pyside2'
+        FALLBACK_API = 'pyside2'
+        warnings.warn(_fallback_fmt.format(API, FALLBACK_API), RuntimeWarning)
```
Comment (ccordoba12) [inline qtpy/__init__.py]: I don't agree with displaying a warning for each binding because it'd make importing qtpy very verbose.

Instead, I think we should do this only if `_EXPLICIT_API is not None` and warn that a new binding `foo` is tried because `_EXPLICIT_API` couldn't be found.

File: qtpy/__init__.py [hunk 3]
```diff
@@ -205,7 +225,16 @@ class PythonQtWarning(Warning):
...
     except ImportError:
-        raise PythonQtError('No Qt bindings could be found')
+        warnings.warn('The binding "{}" could not be found, '
+                      'cannot fall back'.format(API), RuntimeWarning)
+        if _EXPLICIT_API:
+            raise PythonQtError((
+                'No Qt bindings could be found for {}={}, which was '
+                'explicitly requested. Try unsetting the {!r} environ '
+                'or explicitly setting it to a different valid API: {!r}'
+            ).format(QT_API, _EXPLICIT_API, QT_API, VALID_APIS))
```
Comment (ccordoba12) [inline qtpy/__init__.py]: With my suggested change above, I think there's no need to raise this error.

**Comment by ccordoba12** [comment]
Closing due to lack of response.
